### PR TITLE
Start renaming preview1 to p1 and preview2 to p2

### PR DESCRIPTION
**Pull request by alexcrichton**
This is an initial start at renaming the "preview" terminology in WASI targets to "pX". For example the `wasm32-wasi` target should transition to `wasm32-wasip1`, `wasm32-wasi-preview2` should transition to `wasm32-wasip2`, and `wasm32-wasi-threads` should transition to `wasm32-wasip1-threads`. This commit applies a few renames in the `Makefile` such as:

* `WASI_SNAPSHOT` is now either "p1" or "p2"
* The default p2 target triple is now `wasm32-wasip2` instead of `wasm32-wasi-preview2` (in the hopes that it's early enough to change the default).
* Bindings for WASIp2 were renamed from "preview2" terminology to "wasip2".
* The expected-defines files are renamed and the logic of which expectation was used has been updated slightly.

With this commit the intention is that non-preview2 defaults do not change. For example the default build still produces a `wasm32-wasi` sysroot. If `TARGET_TRIPLE=wasm32-wasip1` is passed, however, then that sysroot is produced instead. Similarly a `THREAD_MODEL=posix` build produces a `wasm32-wasi-threads` sysroot target but you can now also pass `TARGET_TRIPLE=wasm32-wasip1-threads` to rename the sysroot.

My hope is to integrate this into the wasi-sdk repository and build a dual sysroot for these new targets for a release or two so both are supported and then in the future the defaults can be switched away from `wasm32-wasi` to `wasm32-wasip1` as built-by-default.

### CODE REVIEW

**Review by sunfishcode — APPROVED** [review]

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -86,15 +86,18 @@ jobs:
       shell: bash
       run: |
         make -j4
-        WASI_SNAPSHOT=preview2 make -j4
+        make -j4 TARGET_TRIPLE=wasm32-wasip1
+        WASI_SNAPSHOT=p2 make -j4
```
[the remark this answers is not in the document]
Comment (alexcrichton) [reply .github/workflows/main.yml]: Sure yeah, in lieu of comments I went ahead and added `TARGET_TRIPLE` for all of the invocations so it's more clear at-a-glance which each one is doing

**Comment by yamt** [comment]
> My hope is to integrate this into the wasi-sdk repository and build a dual sysroot for these new targets for a release or two so both are supported and then in the future the defaults can be switched away from wasm32-wasi to wasm32-wasip1 as built-by-default.

the name "p1" "p2" sound rather cryptic to me.
what's the motivation of the rename?

also, do you have any non-aesthetic reasons to rename preview1?
otherwise, i'm against it.

it's a bit concerning that this kind of change lands almost immediately.

**Comment by sunfishcode** [comment]
The "p2" naming was discussed [at the WASI-02-20 meeting](https://github.com/WebAssembly/meetings/blob/main/wasi/2024/WASI-02-22.md).

**Comment by yamt** [comment]
> The "p2" naming was discussed [at the WASI-02-20 meeting](https://github.com/WebAssembly/meetings/blob/main/wasi/2024/WASI-02-22.md).

thank you. i will take a look.

**Pull request merged**